src/test/java/org/minions/devfund/yuri/WackAMoleTest.java

```diff
@@ -0,0 +1,229 @@
+     * Verifies that the moles left decrease when a moles is hit.
+     */
+    @Test
+    public void testMolesLeftDecreaseWhenAMoleIsWhacked() {
+        final int attempts = 5;
+        final int dimension = 10;
+        WhackAMole game = new WhackAMole(attempts, dimension);
+        game.place(POS_X, POS_Y);
+        game.whack(POS_X, POS_Y);
+        assertEquals(0, game.getMolesLeft());
+    }
+
+    /**
+     * Verifies that the cells grid are filled.
+     */
+    @Test
+    public void testAllCellsGridAreFilled() {
+        final int attempts = 5;
+        final int dimension = 10;
+        WhackAMole game = new WhackAMole(attempts, dimension);
+        int accumulator = 0;
+        for (int row = 0; row < game.getDimension(); row++) {
+            for (int column = 0; column < game.getMoleGrid()[row].length; column++) {
+                if (!Character.isWhitespace(game.getMoleGrid()[row][column])) {
+                    accumulator++;
+                }
+            }
+        }
+        assertEquals(CELLS_AMOUNT, accumulator);
+    }
+
+    /**
+     * Verifies that  grid for user is not empty.
+     */
+    @Test
+    public void testPrintGridToUser() {
+        String expectedResult = " ";
+        assertNotEquals("The grid is empty", expectedResult, game.printGridToUser());
+    }
+
+    /**
+     * Verifies that  grid is not empty.
+     */
+    @Test
+    public void testPrintGrid() {
+        String expectedResult = " ";
+        assertNotEquals("The grid is empty", expectedResult, game.printGrid());
+    }
+
+    /**
+     * Verifies that initial score for a player is  cero.
+     */
+    @Test
+    public  void testInitialScore() {
+        int expectedResult = 0;
+        assertEquals(expectedResult, game.getScore());
+    }
+
+    /**
+     * Verifies that is possible to hit to empty cell.
+     */
+    @Test
+    public void testWhackToEmptyCell() {
+        int posX = 0;
+        int posY = 0;
+        game.getMoleGrid()[posX][posY] = '*';
+        game.whack(posX, posY);
+        assertEquals('W', game.getMoleGrid()[posX][posY]);
+    }
+
+    /**
+     * Verifies that a cell contains a mole.
+     */
+    @Test
+    public void testTheCellIsMole() {
+        int posX = 0;
+        int posY = 0;
+        game.place(posX, posY);
+        char expectedResult = 'M';
+        assertEquals(expectedResult, game.getMoleGrid()[posX][posY]);
+    }
+
+    /**
+     * Verifies that a player is able to win.
+     */
+    @Test
+    public void testPlayerWin() {
+        final int attempts = 5;
+        final int dimension = 10;
+        WhackAMole game = new WhackAMole(attempts, dimension);
+        game.place(0, 0);
+        game.whack(0, 0);
+        assertTrue(game.playerWin());
+    }
+
+    /**
+     * Verifies that a player is able to win.
+     */
+    @Test
+    public void testPlayerLose() {
+        final int attempts = 5;
+        final int dimension = 10;
+        WhackAMole game = new WhackAMole(attempts, dimension);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "game" which hides the field declared at line 24. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AHiddenFieldCheck)